app/services/orchestration_template_dialog_service.rb

```diff
@@ -213,7 +213,7 @@ def add_parameter_field(parameter, group, position)
   end
 
   def create_parameter_dropdown_list(parameter, group, position, dropdown)
-    dropdown_list = dropdown.allowed_values.collect { |v| [v, v] }
+    dropdown_list = dropdown.allowed_values.collect { |v| v.kind_of?(Hash) ? [v.keys.first, v[v.keys.first]] : [v, v] }
```

Comment:
the `dropdown` argument can be either a hash or an array, so the converted dropdown_list should be
```
values = dropdown.allowed_values
dropdown_list = values.kind_of?(Hash) ? values.to_a : values.collect { |v| [v, v] } 
```